magenta-lib/src/main/scala/magenta/tasks/FastlyComputeTasks.scala

```diff
@@ -0,0 +1,185 @@
+
+      uploadPackage(
+        nextVersionNumber,
+        s3Package,
+        client,
+        resources.reporter,
+        stopFlag
+      )
+
+      activateVersion(nextVersionNumber, client, resources.reporter, stopFlag)
+
+      resources.reporter
+        .info(
+          s"Fastly Compute@Edge service ${client.serviceId} - version $nextVersionNumber is now active"
+        )
+    }
+  }
+
+  def stopOnFlag[T](stopFlag: => Boolean)(block: => T): T =
+    if (!stopFlag) block
+    else
+      throw DeployStoppedException(
+        "Deploy manually stopped during UpdateFastlyPackage"
+      )
+
+  private def getActiveVersionNumber(
+      client: FastlyApiClient,
+      reporter: DeployReporter,
+      stopFlag: => Boolean
+  ): Int = {
+    stopOnFlag(stopFlag) {
+      val versionList = block(client.versionList())
+      val versions = Json.parse(versionList.getResponseBody).as[List[Version]]
+      val activeVersion = versions.filter(x => x.active.getOrElse(false)).head
+      reporter.info(s"Current active version ${activeVersion.number}")
+      activeVersion.number
+    }
+  }
+
+  private def clone(
+      versionNumber: Int,
+      client: FastlyApiClient,
+      reporter: DeployReporter,
+      stopFlag: => Boolean
+  ): Int = {
+    stopOnFlag(stopFlag) {
+      val cloned = block(client.versionClone(versionNumber))
+      val clonedVersion = Json.parse(cloned.getResponseBody).as[Version]
+      reporter.info(s"Cloned version ${clonedVersion.number}")
+      clonedVersion.number
+    }
+  }
+
+  /** A helper method that creates a `java.io.File` from an S3 object, making
+    * sure that it has the correct Compute@Edge package extension.
+    * @param obj
+    *   The S3 Object to be converted to an instance of the `java.io.File` class
+    * @param reporter
+    *   The Deploy Reporter
+    * @return
+    *   a valid File
+    */
+  private def createFileFromS3Object(
+      obj: S3Object,
+      reporter: DeployReporter
+  ): File = {
+    if (!obj.extension.contains("gz")) {
+      reporter.fail("The object is not a valid Compute@Edge package")
+    }
+
+    val fileName = obj.relativeTo(s3Package)
+
+    val getObjectRequest = GetObjectRequest
+      .builder()
+      .bucket(obj.bucket)
+      .key(obj.key)
+      .build()
+
+    reporter.info(s"About to fetch $fileName from S3")
+    val `package` =
+      withResource(artifactClient.getObject(getObjectRequest)) { stream =>
+        // `fromInputStream` uses an implicit codec which needs to be
+        // overridden in order to convert binary to Latin-1, hence ISO-8859-1.
+        // This will allow us to send a valid HTTP request to the Fastly API
+        val bufferedSource =
+          scala.io.Source.fromInputStream(stream)(ISO8859)
+        val bytes =
+          bufferedSource.mkString.getBytes(StandardCharsets.ISO_8859_1)
+        Files
+          .write(
+            Files
+              .createTempFile(fileName.replace(".tar.gz", ""), ".tar.gz"),
```

Comment:
`fileName` is a string which includes the extension (eg `package-name.tar.gz`), and `createTempFile`'s constructor takes a prefix and a suffix. If I don't remove it, the file will be called `package-name.tar.gz.tar.gz`. 
I don't think I need to do this at all though, since both `prefix` and `suffix` may be `null`, but it just looks nicer in the logs...
<img width="641" alt="image" src="https://user-images.githubusercontent.com/57295823/235928268-49abac08-f5b2-4385-9c79-e655a029bb6d.png">

Reply:
Ah! Might be worth adding this as an inline comment here?